crates/subspace-service/src/sync_from_dsn/segment_header_downloader.rs

```diff
@@ -96,7 +98,7 @@ impl<'a> SegmentHeaderDownloader<'a> {
     async fn get_last_segment_header(
         &self,
     ) -> Result<Option<(SegmentHeader, Vec<PeerId>)>, Box<dyn Error>> {
-        let mut last_peer_blocks = BTreeMap::default();
+        let mut peer_segment_headers = HashMap::<PeerId, Vec<SegmentHeader>>::default();
```

Comment:
why is this needed?

Reply:
Can you clarity the question?

Reply:
BTreeMap -> HashMap change .. is this because ordering is no longer a concern (as things are  happening in parallel now), or for performance reasons as HashMap gives O(1) operations?

Reply:
`HashMap` is preferred, I'm not exactly sure why we needed or if we needed it at all before, so I have changed it.